feat: Initial commit of CIS module (#349)
terraform-ibm-modules/terraform-ibm-cis: modules/domain/README.md

> # CIS Domain Module

Reviewer: ```suggestion
# CIS domain module
```

> This module is used to manage CIS Domain.

Reviewer: ```suggestion
This module is used to manage CIS Domain.
```
```suggestion
This module is used to manage the IBM Cloud Internet Services (CIS) domain.
```

> This module is used to manage CIS Domain.

Reviewer 2: Perhaps a wording along the lines of "configuring the domain that is assigned to this CIS instance" to distinguish from the DNS subdomain registration (in dns module)

Author: Made some changes, please check.

**Note:**

> After successful execution of this module, the domain configured in the CIS instance will be in `pending` state and will become `active` after the authoritative name servers assigned to the domain are configured correctly at the DNS provider or registrar. The information regarding the name servers assigned can be found in the output of this module as variable `name_servers`. This step involves manual invervention and once the configuration at the DNS registrar is completed, its status moves to `active`. Refer [this](https://cloud.ibm.com/docs/cis?topic=cis-domain-lifecycle-concepts) to get more information.

Reviewer: ```suggestion
After this module run successfully, the status of the domain that is configured in the CIS instance is set to `pending`. You then configure the name servers that are assigned to the domain.

You can find the assigned name servers in the `name_servers` variable of the module output.

The status changes to `active` after the name servers are configured correctly at the DNS provider or registrar. 

For more information, see [Domain lifecycle concepts](https://cloud.ibm.com/docs/cis?topic=cis-domain-lifecycle-concepts).
```

make sure this is still accurate

Author: done
